requirements/dev-requirements.txt

```diff
@@ -747,7 +754,7 @@ sphinxcontrib-serializinghtml==1.1.5
     # via sphinx
 sqlagg==0.17.2
     # via -r base-requirements.in
-sqlalchemy==1.3.19
```

Comment:
Looks like this is a breaking change.
`sqlagg` uses a method called `append_group_by` here:

https://github.com/dimagi/sql-agg/blob/98d20c177087272b57d525ab58facbe6b7f979f1/sqlagg/base.py#L164

As of sqlalchemy 1.4, this was deprecated:
https://github.com/sqlalchemy/sqlalchemy/blob/rel_1_4_49/lib/sqlalchemy/sql/selectable.py#L3425

and it looks to have been removed in the 2.0 release line:
https://github.com/sqlalchemy/sqlalchemy/blob/rel_2_0_19/lib/sqlalchemy/sql/selectable.py

This causes a hard failure restoring users that have mobile UCRs configured: 

```
  <full trace omitted>
  File "/home/ethan/commcare-hq/corehq/apps/userreports/sql/data_source.py", line 64, in get_data
    ret = super(ConfigurableReportSqlDataSource, self).get_data(start=start, limit=limit)
  File "/home/ethan/commcare-hq/corehq/apps/reports/sqlreport.py", line 272, in get_data
    data = self._get_data(start=start, limit=limit)
  File "/home/ethan/commcare-hq/corehq/apps/reports/sqlreport.py", line 288, in _get_data
    return qc.resolve(session.connection(), self.filter_values)
  File "/home/ethan/.pyenv/versions/hq/lib/python3.9/site-packages/sqlagg/base.py", line 291, in resolve
    result = qm.execute(self.connection, filter_values or {})
  File "/home/ethan/.pyenv/versions/hq/lib/python3.9/site-packages/sqlagg/base.py", line 100, in execute
    query = self._build_query()
  File "/home/ethan/.pyenv/versions/hq/lib/python3.9/site-packages/sqlagg/base.py", line 139, in _build_query
    return self._build_query_generic(
  File "/home/ethan/.pyenv/versions/hq/lib/python3.9/site-packages/sqlagg/base.py", line 164, in _build_query_generic
    query.append_group_by(column(group_key))
AttributeError: 'Select' object has no attribute 'append_group_by'
```

Reply:
update: it looks like it also breaks if you attempt to view the UCR in HQ.  The UCR I'm testing with is a trivial one built using report builder - no aggregation, just copying properties over from the case